modules/javafx.graphics/src/main/java/javafx/application/Platform.java

```diff
@@ -432,4 +436,260 @@ public static ReadOnlyBooleanProperty accessibilityActiveProperty() {
+        /**
+         * The platform appearance, which specifies whether applications should use a light
+         * or dark color scheme. The value of this property is derived from the perceptual
+         * brightness of {@link #backgroundColorProperty() backgroundColor} in relation to
+         * {@link #foregroundColorProperty() foregroundColor} and defaults to {@link Appearance#LIGHT}
+         * if the platform does not report color preferences.
+         *
+         * @return the {@code appearance} property
+         */
+        ReadOnlyObjectProperty<Appearance> appearanceProperty();
+
+        Appearance getAppearance();
+
+        /**
+         * The color used for background regions.
+         * <p>
+         * If the platform does not report a background color, this property defaults to {@code Color.WHITE}.
+         *
+         * @return the {@code backgroundColor} property
+         * @defaultValue {@link Color#WHITE}
+         */
+        ReadOnlyObjectProperty<Color> backgroundColorProperty();
+
+        Color getBackgroundColor();
+
+        /**
+         * The color used for foreground elements like text.
+         * <p>
+         * If the platform does not report a foreground color, this property defaults to {@code Color.BLACK}.
+         *
+         * @return the {@code foregroundColor} property
+         * @defaultValue {@code Color.BLACK}
+         */
+        ReadOnlyObjectProperty<Color> foregroundColorProperty();
+
+        Color getForegroundColor();
+
+        /**
+         * The accent color.
+         * <p>
+         * If the platform does not report an accent color, this property defaults to {@code #157EFB}.
+         *
+         * @return the {@code accentColor} property
+         * @defaultValue {@code #157EFB}
+         */
+        ReadOnlyObjectProperty<Color> accentColorProperty();
+
+        Color getAccentColor();
+
+        /**
+         * Returns the {@code Integer} instance to which the specified key is mapped.
+         *
+         * @param key the key
+         * @throws NullPointerException if {@code key} is null
+         * @throws IllegalArgumentException if the key is not mapped to a {@code Integer} instance
+         * @return the {@code Integer} instance to which the key is mapped, or {@code Optional.empty()}
+         *         if no mapping exists for the specified key
+         */
+        Optional<Integer> getInteger(String key);
+
+        /**
+         * Returns the {@code Double} instance to which the specified key is mapped.
+         *
+         * @param key the key
+         * @throws NullPointerException if {@code key} is null
+         * @throws IllegalArgumentException if the key is not mapped to a {@code Double} instance
+         * @return the {@code Double} instance to which the key is mapped, or {@code Optional.empty()}
+         *         if no mapping exists for the specified key
+         */
+        Optional<Double> getDouble(String key);
```

Comment:
1. There is no `Double` value now, but there might be in the future. For exampe, the API may expose system-provided double-click times, or it may expose information about the dimensions of system decorations. As for `Paint`, it might be conceivable that a platform would expose color gradients.

2. The list is only fully known for the three listed platforms, but it's unknown for other platforms. Using an enum key instead of a string key is certainly possible, but then we'd be hard-coding platform-specific constants into the core of JavaFX. You might argue that providing a list of supported keys isn't really all that different from using a language feature like enums to represent the constants. Using platform-specific preferences requires a deep understanding of the platform in any case, and is not something application developers should rely on regularly. Instead, they should be using the common subset of well-known preferences, which is easily accessible via the properties on the `Platform` class.

Reply:
Alright, my assumption that the listed keys are complete is wrong then. I thought that "The following list contains all preferences that are potentially available on the specified platforms" was a good enough guarantee.